src/components/ImportAction.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { useDispatch } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { useHistory, useLocation } from 'react-router-dom';
+import { Button, ButtonVariant, Tooltip } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
+import { closeOLS, importCodeblock } from '../redux-actions';
+import ConfirmationModal from './ConfirmationModal';
+
+type Props = {
+  value: string;
+};
+
+const ImportAction: React.FC<Props> = ({ value }) => {
+  const { t } = useTranslation('plugin__lightspeed-console-plugin');
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const [activeNamespace] = useActiveNamespace();
+
+  const isImportYAMLPage = location?.pathname?.split('/').pop() === 'import';
+
+  const triggerRef = React.createRef<HTMLButtonElement>();
+  const [showModal, setShowModal] = React.useState(false);
+
+  const generateRandomNumericString = (length = 4): string => {
+    const digits = '0123456789';
+    return Array.from({ length }, () => digits[Math.floor(Math.random() * digits.length)]).join('');
+  };
+
+  const handleRedirect = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (event.currentTarget.id === 'leave') {
+      history.push(`/k8s/ns/${activeNamespace}/import?ols=true`);
+    }
+    setShowModal(false);
+    dispatch(closeOLS());
```

Comment:
I think we should only close the OLS UI if the "Leave" button is clicked (don't close it if you click the "Stay" button.